plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMetadata.java

```diff
@@ -661,11 +649,9 @@ public Iterator<TableColumnsMetadata> streamTableColumns(ConnectorSession sessio
                         }
                         String tableLocation = metastoreTable.get().location();
                         MetadataEntry metadata = transactionLogAccess.getMetadataEntry(getSnapshot(table, tableLocation, session), session);
-                        Map<String, String> columnComments = getColumnComments(metadata);
-                        Map<String, Boolean> columnsNullability = getColumnsNullability(metadata);
-                        Map<String, String> columnGenerations = getGeneratedColumnExpressions(metadata);
+                        DeltaTable deltaTable = DeltaTable.of(metadata);
                         List<ColumnMetadata> columnMetadata = getColumns(metadata).stream()
-                                .map(column -> getColumnMetadata(column, columnComments.get(column.getColumnName()), columnsNullability.getOrDefault(column.getBaseColumnName(), true), columnGenerations.get(column.getBaseColumnName())))
+                                .map(column -> getColumnMetadata(column, deltaTable.getColumnComment(column.getColumnName()), deltaTable.getNullability(column.getBaseColumnName()), deltaTable.getColumnGeneration(column.getBaseColumnName())))
```

Comment:
Is this a preparatory commit for #18013 ?

The current approach still doesn't solve the issue of being able to deal with case insensitive column names.

Reply:
> Is this a preparatory commit for https://github.com/trinodb/trino/issues/18013 ?

Yes. 

> The current approach still doesn't solve the issue

That's the expected behavior.